index.js

```diff
@@ -96,89 +103,72 @@ function myMap(array, callback) {
-// console.log("%cCallbacks - Question 1", "color: red")
-
-// console.log(myMap([1,2,3,4,5], triple)) 
+console.log("%cCallbacks - Question 1", "color: red")
+const triple = (arr) => {
+  return arr * 3;
+}
+console.log(myMap([1,2,3,4,5], triple))
 // // => [3,6,9,12,15]
-
-// console.log(myMap([2,4,6,8], triple))   
+console.log(myMap([2,4,6,8], triple))
 // // => [6,12,18,24]
-// console.log("%c----------", "color: red") 
-
-
+console.log("%c----------", "color: red")
 // ***** Callbacks - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cCallbacks - Question 2", "color: red")
-
-// console.log(myMap(["Raffy", "Chase"], greet)) 
+console.log("%cCallbacks - Question 2", "color: red")
+const greet = (arr) => {
+  return `Heyyy, ${arr}`
+}
+console.log(myMap(["Raffy", "Chase"], greet))
 // // => ["Hi, Raffy!", "Hi, Chase!"]
-// console.log("%c----------", "color: red") 
-
-
-
+console.log("%c----------", "color: red")
 // ***** Scope & Closures *****
-
 function takeANumber(line, name) {
   line.push(name)
-
   return `Welcome, ${name}. You are number ${line.length} in line.`
 }
-
 function nowServing(line) {
   if (!line.length) {
     return "There is nobody waiting to be served!"
   }
   return `Currently serving ${line.shift()}.`
 }
-
 const line = []
-
 // ***** Scope & Closures - Question 1 *****
-
-
 // *** Uncomment the lines below to test
 // console.log("%cScope & Closures - Question 1", "color: red")
-
-// console.log(takeATicketNumber(line))
+// function takeTickerNumber(line) {
+//   ticket += 1;
+//   line.push(ticket)
+//   return `Welcome. You are ticket number ${ticket}`
+// }
+// console.log(takeTickerNumber(line))
 // // => `Welcome. You are ticket number 1`
-
-// console.log(takeATicketNumber(line))
+// console.log(takeTickerNumber(line))
 // // => `Welcome. You are ticket number 2`
-
 // console.log(nowServing(line))
 // // => `Currently serving 1.`
-
 // console.log(nowServing(line))
 // // => `Currently serving 2.`
-
-// console.log(takeATicketNumber(line))
+// console.log(takeTickerNumber(line))
 // // => `Welcome. You are ticket number 3`
-
-// console.log("%c----------", "color: red") 
-
+// console.log("%c----------", "color: red")
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
-
+console.log("%cScope & Closures - Question 2", "color: red")
 // const newLine = []
 // const takeATicketNumberFunc = ticketNumberGeneratorFunc()
 // console.log(takeATicketNumberFunc(newLine))
 // // => `Welcome. You are ticket number 1`
-
 // console.log(takeATicketNumberFunc(newLine))
 // // => `Welcome. You are ticket number 2`
-
 // console.log(nowServing(newLine))
 // // => `Currently serving 1.`
-
 // console.log(nowServing(newLine))
 // // => `Currently serving 2.`
-
 // console.log(takeATicketNumberFunc(newLine))
 // // => `Welcome. You are ticket number 3`
+// console.log("%c----------", "color: red")
```

Comment:
This could be solved in the following way

`const ticketNumberGeneratorFunc = () => {
  let number = 0;

  return (line) => {
    number += 1;
    line.push(number);
    return `Welcome. You are ticket number ${number}`;
  }
}`

It is basically just encapsulating all the logic from the previous question into one function.